TheMovieDb/MovieDetail/Controller/DetailViewController.swift.swift

```diff
@@ -0,0 +1,188 @@
+        configureUI()
+        fetchDataAPI()
+    }
+    
+    private func fetchDataAPI() {
+        fetchSimilar()
+        fetchRecommendation()
+        group.notify(queue: .main) {
+            self.reloadCollectionView()
+        }
+    }
+        
+    init(with movie: Movie) {
+        self.movie = movie
+        let layout = UICollectionViewFlowLayout()
+        layout.sectionInset = UIEdgeInsets(top: 0, left: 10, bottom: 0, right: 10)
+        layout.itemSize = CGSize(width: 100, height: 140)
+        super.init(collectionViewLayout: layout)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - API
+    
+    private func fetchSimilar() {
+        group.enter()
+        MovieAPI.shared.getRecommendations(completion: {(response: Result<Movies, Error>) in
+            switch response {
+            case .failure(let error):
+                debugPrint(error)
+            case .success(let res):
+                self.recommendationsMovies = res.movies
+            }
+            self.group.leave()
+        })
+    }
+    
+    private func fetchRecommendation() {
+        group.enter()
+        MovieAPI.shared.getSimilar(completion: {(response: Result<Movies, Error>) in
+            switch response {
+            case .failure(let error):
+                debugPrint(error)
+            case .success(let res):
+                self.similarMovies = res.movies
+                // self.reloadCollectionView(groupSection: .topRated)
+            }
+            self.group.leave()
+        })
+    }
+    
+    private func reloadCollectionView() {
+        DispatchQueue.main.async {
+            self.collectionView.reloadData()
+        }
+    }
+    
+    // MARK: - Helpers
+    private func configureUI() {
+        collectionView.register(DetailHeaderView.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: DetailHeaderView.reuseIdentifier)
+        collectionView.register(DetailHeader.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: DetailHeader.reuseIdentifier)
+        
+        collectionView.register(DefaultSectionCell.self, forCellWithReuseIdentifier: DefaultSectionCell.reuseIdentifier)
+        
+        navigationItem.title = movie.title
+    }
+    
+}
+
+// MARK: - UICollectionViewDataSource
+extension DetailViewController {
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        let section = RelatedMovieSections(rawValue: section)
+        
+        switch section {
+        case .recommendations:
+            return recommendationsMovies.count
+        case .similar:
+            return similarMovies.count
+        case .none:
+            return 0
+        }
+        
+    }
+    
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 2
```

Comment:
maybe we can use a model here

Reply:
Now I´m using the enum that has the number of section